### PR TITLE
[arm64ec] Fix Windows toolchain.

### PR DESCRIPTION
**Pull request by BillyONeal**
* set(CMAKE_TRY_COMPILE_TARGET_TYPE STATIC_LIBRARY) when linking arm64ec static to workaround softintrin as explained in the comment.
* Switch to using the _INIT versions of flags now that we require CMake >= 3.7 (as of https://github.com/microsoft/vcpkg/pull/23134 )

### CODE REVIEW

**Comment by BillyONeal** [comment]
I intentionally cancelled the validation build because this triggers a world rebuild and I would like folks to say it isn't entirely broken first.

**Comment by BillyONeal** [comment]
Probably should be "rolled up" with https://github.com/microsoft/vcpkg/pull/22831

File: scripts/toolchains/windows.cmake [hunk 1]
```diff
@@ -9,6 +9,11 @@ if(NOT _CMAKE_IN_TRY_COMPILE)
         set(VCPKG_CRT_LINK_FLAG_PREFIX "/MD")
     elseif(VCPKG_CRT_LINKAGE STREQUAL "static")
         set(VCPKG_CRT_LINK_FLAG_PREFIX "/MT")
+        if (VCPKG_TARGET_ARCHITECTURE STREQUAL "arm64ec")
+            # When linking statically to the CRT, an executable also has to link against softintrin.lib, but that
```
Comment (ras0219-msft) [inline scripts/toolchains/windows.cmake:13]: > an executable also has to link against softintrin.lib

So should that be added to the exe link flags?

Comment (BillyONeal) [reply scripts/toolchains/windows.cmake:13]: In Phil's case the lib isn't available in the environment he wants; it's something that needs to be fixed in the product itself.

**Comment by BillyONeal** [comment]
We need to be careful of what happened in: 

* https://github.com/microsoft/vcpkg/pull/9862
* https://github.com/microsoft/vcpkg/pull/10562

**Comment by BillyONeal** [comment]
See also: https://github.com/microsoft/vcpkg/pull/16111 (thanks to Iluin on Discord for pointing these out)

**Comment by Neumann-A** [comment]
>  (thanks to Iluin on Discord for pointing these out)

That is just me on mobile. 

Hmm what is actually the plan behind switching to the `_INIT` variables?

Maybe it is time to have dedicated toolchains instead of one fits it all?

**Comment by BillyONeal** [comment]
> > (thanks to Iluin on Discord for pointing these out)
> 
> That is just me on mobile.

Ha! Well thank you nonetheless :)

> Hmm what is actually the plan behind switching to the `_INIT` variables?

There isn't really so much of a "plan" here; I just applied the change that some arm64ec folks are using because it looked mostly harmless. Your comments have indicated that it isn't mostly harmless so I don't know on what timescale this is going to land. I'm still a bit hazy on what the end game for this arm64ec feature is.

> Maybe it is time to have dedicated toolchains instead of one fits it all?

I have no opinion on such a change.

**Comment by Neumann-A** [comment]
> There isn't really so much of a "plan" here; 

Maybe the people who use it should point out the difference in compiler flags. I only see:
https://gitlab.kitware.com/cmake/cmake/-/blob/master/Modules/Platform/Windows-MSVC.cmake#L229
https://gitlab.kitware.com/cmake/cmake/-/blob/master/Modules/Platform/Windows-MSVC.cmake#L272
https://gitlab.kitware.com/cmake/cmake/-/blob/master/Modules/Platform/Windows-MSVC.cmake#L320

**Pull request closed without merging**